DQM/HLTEvF/plugins/TriggerRatesMonitor.cc

```diff
@@ -191,19 +188,35 @@ void TriggerRatesMonitor::dqmBeginRun(edm::Run const &run,
       histograms.hlt_by_dataset_counts[i].resize(paths.size());
       histograms.datasets[i].reserve(paths.size());
       for (auto const &path : paths) {
-        histograms.datasets[i].push_back(histograms.hltConfig.triggerIndex(path));
+        auto const triggerIdx = histograms.hltConfig.triggerIndex(path);
+        if (triggerIdx < nTriggers)
+          histograms.datasets[i].push_back(triggerIdx);
+        else
+          LogDebug("TriggerRatesMonitor")
+              << "The rates of the HLT path \"" << path << "\" (dataset: \"" << histograms.hltConfig.datasetName(i)
+              << "\") will not be monitored for this run.\nThis HLT path is not available in the process \""
+              << labels.process << "\", but it is listed in its \"datasets\" PSet.";
       }
     }
     histograms.dataset_counts.clear();
     histograms.dataset_counts.resize(datasets);
 
-    unsigned int streams = histograms.hltConfig.streamNames().size();
+    unsigned int const streams = histograms.hltConfig.streamNames().size();
     histograms.streams.clear();
     histograms.streams.resize(streams);
     for (unsigned int i = 0; i < streams; ++i) {
       for (auto const &dataset : histograms.hltConfig.streamContent(i)) {
-        for (auto const &path : histograms.hltConfig.datasetContent(dataset))
-          histograms.streams[i].push_back(histograms.hltConfig.triggerIndex(path));
+        for (auto const &path : histograms.hltConfig.datasetContent(dataset)) {
+          auto const triggerIdx = histograms.hltConfig.triggerIndex(path);
+          if (triggerIdx < nTriggers)
+            histograms.streams[i].push_back(triggerIdx);
```

Comment:
This (i.e. the check on `< nTriggers`) is the change to fix the error reported in [CMSHLT-2577](https://its.cern.ch/jira/browse/CMSHLT-2577). A similar check can be found in
https://github.com/cms-sw/cmssw/blob/7e6539b24427faa92b1411a9903202169a3c3990/HLTrigger/JSONMonitoring/plugins/HLTriggerJSONMonitoring.cc#L230-L232

The rest of the changes in this PR is about adding a few checks, plus minor cleanup.